core/rewards.lua

```diff
@@ -559,20 +559,78 @@ function Transmog:GetStatus()
     return status
 end
 
+-------------------------------------------------------------------------------
+------------------------------------ RECIPE -----------------------------------
+-------------------------------------------------------------------------------
+
+local Recipe = Class('Recipe', Reward, {
+    display_option = 'show_recipe_rewards',
+    type = L['recipe']
+})
+
+function Recipe:Initialize(attrs)
+    if attrs.item then
+        Item.Initialize(self, attrs)
+    else
+        Reward.Initialize(self, attrs)
+        local prefix = {
+            [171] = L['recipe_prefix_alchemy'],
+            [164] = L['recipe_prefix_blacksmithing'],
+            [333] = L['recipe_prefix_enchanting'],
+            [202] = L['recipe_prefix_engineering'],
+            [773] = L['recipe_prefix_inscription'],
+            [755] = L['recipe_prefix_jewelcrafting'],
+            [165] = L['recipe_prefix_leatherworking'],
+            [197] = L['recipe_prefix_tailoring'],
+            [185] = L['recipe_prefix_cooking']
+        }
+        prefix = prefix[self.profession] or ''
+
+        local name, _, icon = GetSpellInfo(self.id)
+        self.itemIcon = icon
+        self.itemLink = ns.color.White('[' .. prefix .. name .. ']')
+    end
+end
+
+function Recipe:IsObtained() return IsSpellKnown(self.id) end
+
+function Recipe:IsEnabled()
+    if self.display_option and not ns:GetOpt(self.display_option) then
+        return false
+    end
+    if self.profession and ns.PlayerHasProfession(self.profession) then
+        return true
+    end
+    return false
+end
+
+function Recipe:GetText()
+    local text = self.itemLink .. ' (' .. self.type .. ')'
+    if self.note then -- additional info
+        text = text .. ' (' .. ns.RenderLinks(self.note, true) .. ')'
+    end
+    return Icon(self.itemIcon) .. text
+end
```

Comment:
Delete this function once we inherit from `Item`. The `Item.GetText()` function will do the same thing.